Lib/multiprocessing/forkserver.py

```diff
@@ -150,22 +150,25 @@ def main(listener_fd, alive_r, preload, main_path=None, sys_path=None):
     util._close_stdin()
 
     sig_r, sig_w = os.pipe()
+    os.set_blocking(sig_r, False)
     os.set_blocking(sig_w, False)
 
     def sigchld_handler(*_unused):
-        try:
-            os.write(sig_w, b'.')
-        except BlockingIOError:
-            pass
```

Comment:
I like this change :-) Using the existing C signal handler for the write is probably more reliable ;-)